src/api/v1/db/oracledb/pizzas-dao.js

```diff
@@ -217,4 +248,104 @@ const getPizzaById = async (pizzaId, query) => {
   }
 };
 
-export { getPizzas, getPizzaById };
+/**
+ * Generate queries and bind params for an ingredients relationship resource collection
+ *
+ * @param {object[]} ingredientRelationshipData the `data` key of
+ * `body.data.relationships.ingredients`
+ * @returns {object[]} an object with `queries` and `bindParams` members
+ */
+const getIngredientQueriesAndBindParams = async (ingredientRelationshipData) => {
+  const bindParams = {};
+  const ingredientIds = ingredientRelationshipData
+    .map((ingredient) => ingredient.id);
+  if (!await checkIngredientsExist(ingredientIds)) {
+    throw new Error('ingredient IDs are invalid');
+  }
+  const queries = ingredientIds.map((id) => {
+    bindParams[`ingredientId${id}`] = parseInt(id, 10);
+    return dedent`
+    INTO PIZZA_INGREDIENTS (INGREDIENT_ID, PIZZA_ID)
+      VALUES (:ingredientId${id}, :pizzaId)
+    `;
+  });
+  return { queries, bindParams };
+};
+
+/**
+ * insert ingredients relationships resources into the
+ * PIZZA_INGREDIENTS table
+ *
+ * @param {string[]} ingredientQueries
+ * @param {object} ingredientBindParams
+ * @param {OracleDB.Connection} connection
+ */
+const insertIngredients = async (ingredientQueries, ingredientBindParams, connection) => {
+  const insertIngredientsQuery = dedent`
+  INSERT ALL
+    ${ingredientQueries.join('\n  ')}
+  SELECT * FROM dual
+  `;
+  await connection.execute(
+    insertIngredientsQuery,
+    ingredientBindParams,
+    { autoCommit: true },
+  );
+};
+/**
+ * Create a pizza record in the database
+ *
+ * @param {object} body
+ * @returns {Promise<object>} the serialized, newly created pizza
+ */
+const postPizza = async (body) => {
+  const pizzaBindParams = getBindParams(body, pizzaProperties, pizzaOutBindParams);
+  const connection = await getConnection();
+  let ingredientQueries = [];
+  let ingredientBindParams = {};
+
+  const hasIngredients = 'relationships' in body.data && 'ingredients' in body.data.relationships;
```

Comment:
You could use `_.has(body, 'data.relationships.ingredients);` here as well. This is nice because the undefined/null checking should be handled by the function.